src/test/java/teammates/test/cases/webapi/GetStudentProfilePictureActionTest.java

```diff
@@ -29,57 +27,170 @@ protected String getRequestMethod() {
+        assertEquals(student1Profile.pictureKey, imageResult.blobKey);
 
-    private void testActionWithBlobKeySuccess() {
-        ______TS("Typical case: using blobkey");
-        loginAsStudent(account.googleId);
+        ______TS("Success case: student passes in incomplete params but still gets his own image");
 
         String[] submissionParams = new String[] {
-                Const.ParamsNames.BLOB_KEY, studentProfileAttributes.pictureKey,
+                Const.ParamsNames.COURSE_ID, student1InCourse1.getCourse(),
+        };
+
+        action = getAction(submissionParams);
+        imageResult = getImageResult(action);
+
+        assertEquals(HttpStatus.SC_OK, imageResult.getStatusCode());
+
+        submissionParams = new String[] {
+                Const.ParamsNames.STUDENT_EMAIL, student1InCourse1.getEmail(),
         };
-        GetStudentProfilePictureAction action = getAction(submissionParams);
-        ImageResult result = getImageResult(action);
 
-        assertEquals(HttpStatus.SC_OK, result.getStatusCode());
-        assertEquals(studentProfileAttributes.pictureKey, result.blobKey);
+        action = getAction(submissionParams);
+        imageResult = getImageResult(action);
+
+        assertEquals(HttpStatus.SC_OK, imageResult.getStatusCode());
+        assertEquals(student1Profile.pictureKey, imageResult.blobKey);
+
+        ______TS("Success case: student gets his teammate's image");
+        StudentAttributes student2InCourse1 = typicalBundle.students.get("student2InCourse1");
+        gaeSimulation.logoutUser();
+        loginAsStudent(student2InCourse1.googleId);
+
+        submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, student1InCourse1.getCourse(),
+                Const.ParamsNames.STUDENT_EMAIL, student1InCourse1.getEmail(),
+        };
+
+        action = getAction(submissionParams);
+        imageResult = getImageResult(action);
+
+        assertEquals(HttpStatus.SC_OK, imageResult.getStatusCode());
+        assertEquals(student1Profile.pictureKey, imageResult.blobKey);
+
+        ______TS("Success case: instructor with privilege views image of his student");
+        gaeSimulation.logoutUser();
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        loginAsInstructor(instructor1OfCourse1.googleId);
+
+        submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, student1InCourse1.getCourse(),
+                Const.ParamsNames.STUDENT_EMAIL, student1InCourse1.getEmail(),
+        };
+
+        action = getAction(submissionParams);
+        imageResult = getImageResult(action);
+
+        assertEquals(HttpStatus.SC_OK, imageResult.getStatusCode());
+        assertEquals(student1Profile.pictureKey, imageResult.blobKey);
+
+        ______TS("Failure case: requested student has no profile picture");
```

Comment:
Let's add a case where unregistered student's profile is requested?

Reply:
Done